Cython/Compiler/PyrexTypes.py

```diff
@@ -2592,6 +2592,46 @@ def find_cpp_operation_type(self, operator, operand_type=None):
             return self.base_type.find_cpp_operation_type(operator, operand_type)
         return None
 
+class CPtrToArraySliceType(CPtrType):
+    def create_to_py_utility_code(self, env):
+        if self.to_py_function is not None:
+            return True
+        if not self.base_type.create_to_py_utility_code(env):
+            return False
+
+        safe_typename = self.base_type.specialization_name()
+
+        to_py_function = "__Pyx_carray_slice_to_py_%s" % safe_typename
+        to_tuple_function = "__Pyx_carray_slice_to_tuple_%s" % safe_typename
+
+        from .UtilityCode import CythonUtilityCode
+        context = {
+            'cname': to_py_function,
+            'to_tuple_cname': to_tuple_function,
+            'base_type': self.base_type,
+        }
+        env.use_utility_code(CythonUtilityCode.load(
+            "carray_slice.to_py", "CConvert.pyx",
+            outer_module_scope=env.global_scope(),  # need access to types declared in module
+            context=context, compiler_directives=dict(env.global_scope().directives)))
+        self.to_tuple_function = to_tuple_function
+        self.to_py_function = to_py_function
+        return True
+
+    def to_py_call_code(self, source_code, start_code, stop_code, result_code, result_type, to_py_function=None):
+        func = self.to_py_function if to_py_function is None else to_py_function
+
+        target_is_tuple = result_type.is_builtin_type and result_type.name == 'tuple'
```

Comment:
Makes total sense to have these two versions, but the tuple case doesn't seem to be exercised in the tests.